### DISCUSSION TITLE
External Authentication Providers

### DISCUSSION THREAD

**Discussion by LorenzoInvernizzi** [payloadcms/payload#3106] [Q&A]
Hi Everyone,

I am wondering if there are examples, or you can suggest how to use external auth providers, either classic social providers (google, facebook, ..) or custom oauth providers. Searching the docs i couldnt find an answers, googling i found this [payload-plugin-oauth](https://github.com/thgh/payload-plugin-oauth) but i am not sure if it is what i am looking for.

Thanks,
Lor

**Comment by LorenzoInvernizzi** [comment]
+ to clarify i intend the usage of external auth provider in my frontend and how to use these with the backend. Example i use next.js for frontend and the session (useSession) for handling authenticated users. If i could retrieve the session and use it in payload would be awesome.

**Reply by uncvrd** [reply]
Did you happen to have any luck with your findings? We have our own authentication using Ory (it's an open source Auth0), would love for users already registered with our Auth backend to be able to use the same credentials when logging in with our Payload app

**Reply by tobidemski** [reply]
Have you found any reliable solution for this. I probably also looking in a solution like this.
